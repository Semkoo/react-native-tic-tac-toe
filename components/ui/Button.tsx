import React, { ReactElement } from 'react';
import {
  TouchableOpacity,
  TouchableOpacityProps,
  ActivityIndicator,
  StyleSheet,
} from 'react-native';
import Text from './Text';

type ButtonProps = {
  title: string;
  loading?: boolean;
} & TouchableOpacityProps;

export default function Button({ title, style, loading, ...props }: ButtonProps): ReactElement {
  return (
    <TouchableOpacity disabled={loading} {...props} style={[styles.button, style]}>
      {loading ? (
        <ActivityIndicator color="#000" />
      ) : (
        <Text style={styles.buttonText}>{title}</Text>
      )}
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#dafaf7',
    paddingVertical: 20,
    paddingHorizontal: 35,
    borderRadius: 30,
  },
  buttonText: {
    fontSize: 18,
    color: '#221a36',
    textAlign: 'center',
  },
});
