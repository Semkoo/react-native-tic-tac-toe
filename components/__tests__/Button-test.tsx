import * as React from 'react';
import renderer from 'react-test-renderer';

import Button from '../ui/Button';

it(`renders correctly`, () => {
  const tree = renderer.create(<Button title="Test" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with onPress', () => {
  const onPressMock = jest.fn();
  const tree = renderer.create(<Button title="Test" onPress={onPressMock} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with onPress and disabled', () => {
  const onPressMock = jest.fn();
  const tree = renderer.create(<Button title="Test" onPress={onPressMock} disabled />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with onPress and loading', () => {
  const onPressMock = jest.fn();
  const tree = renderer.create(<Button title="Test" onPress={onPressMock} loading />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with onPress and disabled and loading', () => {
  const onPressMock = jest.fn();
  const tree = renderer
    .create(<Button title="Test" onPress={onPressMock} disabled loading />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
