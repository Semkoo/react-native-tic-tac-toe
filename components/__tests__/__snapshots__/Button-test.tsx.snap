// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#dafaf7",
      "borderRadius": 30,
      "opacity": 1,
      "paddingHorizontal": 35,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceMono",
        },
        {
          "color": "#221a36",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`renders correctly with onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#dafaf7",
      "borderRadius": 30,
      "opacity": 1,
      "paddingHorizontal": 35,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceMono",
        },
        {
          "color": "#221a36",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`renders correctly with onPress and disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#dafaf7",
      "borderRadius": 30,
      "opacity": 1,
      "paddingHorizontal": 35,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "SpaceMono",
        },
        {
          "color": "#221a36",
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`renders correctly with onPress and disabled and loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#dafaf7",
      "borderRadius": 30,
      "opacity": 1,
      "paddingHorizontal": 35,
      "paddingVertical": 20,
    }
  }
>
  <ActivityIndicator
    color="#000"
  />
</View>
`;

exports[`renders correctly with onPress and loading 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#dafaf7",
      "borderRadius": 30,
      "opacity": 1,
      "paddingHorizontal": 35,
      "paddingVertical": 20,
    }
  }
>
  <ActivityIndicator
    color="#000"
  />
</View>
`;
